src/report/mod.rs

```diff
@@ -123,7 +123,14 @@ fn write_logs<DB: ReadResults, W: ReportWriter>(
             let log_path = crate_to_path_fragment(krate).join("log.txt");
             let content = db.load_log(ex, tc, krate)
                 .and_then(|c| c.ok_or_else(|| "missing logs".into()))
-                .chain_err(|| format!("failed to read log of {} on {}", krate, tc.to_string()))?;
+                .chain_err(|| format!("failed to read log of {} on {}", krate, tc.to_string()));
+            let content = match content {
+                Ok(c) => c,
+                Err(e) => {
+                    println!("Error while writing log for {}: {}", krate, e);
```

Comment:
Can you use `util::report_error` here? The function also shows error causes, while your code does not (and that means it doesn't show any meaningful error since you used `chain_err`.